pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -45,6 +49,41 @@ func (ms *MetricSender) ReportNetworkDeviceMetadata(config *checkconfig.CheckCon
 		}
 		ms.sender.EventPlatformEvent(string(payloadBytes), epforwarder.EventTypeNetworkDevicesMetadata)
 	}
+
+	// Telemetry
+	for _, interfaceStatus := range interfaces {
+		status := ComputeInterfaceStatus(interfaceStatus.AdminStatus, interfaceStatus.OperStatus)
+		interfaceTags := append(tags, fmt.Sprintf("status:%s", status), fmt.Sprintf("interface:%s", interfaceStatus.Name), fmt.Sprintf("interface_alias:%s", interfaceStatus.Alias))
+		ms.sender.Gauge(interfaceStatusMetric, 1, "", interfaceTags)
+	}
+}
+
+func ComputeInterfaceStatus(adminStatus int32, operStatus int32) string {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏

1/ I think this function should be unit tested for all code paths.

2/ I think there is no need to export the function (Title case) since it's only used in this package.